Fails to build macOS shared lib in 1.18.0, 1.19.0, main (075e319)

# Bug Report

### Is the issue related to model conversion?
No

### Describe the bug
In 1.18.0, 1.19.0 and current main branch (075e319), building shared ONNX library on macOS has failed with undefined symbols.

Example output snippet:
```
Undefined symbols for architecture arm64:
  "onnx::GraphProto::InternalSwap(onnx::GraphProto*)", referenced from:
      onnx::MakeAttribute(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, onnx::GraphProto) in attr_proto_util.cc.o
      onnx::MakeAttribute(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<onnx::GraphProto, std::__1::allocator<onnx::GraphProto>>) in attr_proto_util.cc.o
  "onnx::GraphProto::CopyFrom(onnx::GraphProto const&)", referenced from:
      onnx::MakeAttribute(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, onnx::GraphProto) in attr_proto_util.cc.o
      onnx::MakeAttribute(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::vector<onnx::GraphProto, std::__1::allocator<onnx::GraphProto>>) in attr_proto_util.cc.o
      onnx::OpSchema::Attr(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, onnx::AttributeProto_AttributeType, onnx::GraphProto const&) in schema.cc.o
      onnx::OpSchema::Attr(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, onnx::AttributeProto_AttributeType, std::__1::vector<onnx::GraphProto, std::__1::allocator<onnx::GraphProto>> const&) in schema.cc.o
  "onnx::GraphProto::GraphProto(google::protobuf::Arena*)", referenced from:
      void* google::protobuf::Arena::DefaultConstruct<onnx::GraphProto>(google::protobuf::Arena*) in ir_pb_converter.cc.o
      void* google::protobuf::Arena::DefaultConstruct<onnx::GraphProto>(google::protobuf::Arena*) in ir_pb_converter.cc.o
      onnx::BuildSequenceMapBodyFunc(onnx::FunctionBodyBuildContext const&, onnx::OpSchema const&, onnx::FunctionProto&) in defs.cc.o
  "onnx::GraphProto::GraphProto(google::protobuf::Arena*, onnx::GraphProto const&)", referenced from:
      onnx::BuildSequenceMapBodyFunc(onnx::FunctionBodyBuildContext const&, onnx::OpSchema const&, onnx::FunctionProto&) in defs.cc.o
  "onnx::GraphProto::~GraphProto()", referenced from:
      onnx::BuildSequenceMapBodyFunc(onnx::FunctionBodyBuildContext const&, onnx::OpSchema const&, onnx::FunctionProto&) in defs.cc.o
      onnx::BuildSequenceMapBodyFunc(onnx::FunctionBodyBuildContext const&, onnx::OpSchema const&, onnx::FunctionProto&) in defs.cc.o
      onnx::BuildSequenceMapBodyFunc(onnx::FunctionBodyBuildContext const&, onnx::OpSchema const&, onnx::FunctionProto&) in defs.cc.o
      onnx::BuildSequenceMapBodyFunc(onnx::FunctionBodyBuildContext const&, onnx::OpSchema const&, onnx::FunctionProto&) in defs.cc.o
      onnx::BuildSequenceMapBodyFunc(onnx::FunctionBodyBuildContext const&, onnx::OpSchema const&, onnx::FunctionProto&) in defs.cc.o
  "onnx::ModelProto::Clear()", referenced from:
      onnx::OnnxParser::Parse(onnx::ModelProto&) in parser.cc.o
  "onnx::ModelProto::ModelProto(google::protobuf::Arena*)", referenced from:
      onnx::checker::check_model(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool, bool) in checker.cc.o
      onnx::PrepareOutput(onnx::ModelProto const&) in ir_pb_converter.cc.o
      std::__1::__function::__func<onnx::inliner::(anonymous namespace)::InlinerImpl::Process(google::protobuf::RepeatedPtrField<onnx::NodeProto>&, google::protobuf::RepeatedPtrField<onnx::ValueInfoProto>&)::'lambda'(onnx::NodeProto&), std::__1::allocator<onnx::inliner::(anonymous namespace)::InlinerImpl::Process(google::protobuf::RepeatedPtrField<onnx::NodeProto>&, google::protobuf::RepeatedPtrField<onnx::ValueInfoProto>&)::'lambda'(onnx::NodeProto&)>, void (onnx::NodeProto&)>::operator()(onnx::NodeProto&) in inliner.cc.o
      onnx::shape_inference::InferShapes(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, onnx::ISchemaRegistry const*, onnx::ShapeInferenceOptions const&, std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, onnx::TensorShapeProto, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const, onnx::TensorShapeProto>>>*) in implementation.cc.o
  "onnx::ModelProto::ModelProto(google::protobuf::Arena*, onnx::ModelProto const&)", referenced from:
      onnx::checker::check_model(onnx::ModelProto const&, bool, bool, bool) in checker.cc.o
  "onnx::ModelProto::~ModelProto()", referenced from:
      onnx::checker::check_model(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool, bool) in checker.cc.o
      onnx::checker::check_model(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool, bool) in checker.cc.o
      onnx::checker::check_model(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool, bool) in checker.cc.o
      onnx::checker::check_model(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool, bool) in checker.cc.o
      onnx::checker::check_model(onnx::ModelProto const&, bool, bool, bool) in checker.cc.o
      onnx::checker::check_model(onnx::ModelProto const&, bool, bool, bool) in checker.cc.o
      onnx::PrepareOutput(onnx::ModelProto const&) in ir_pb_converter.cc.o
      onnx::PrepareOutput(onnx::ModelProto const&) in ir_pb_converter.cc.o
      onnx::PrepareOutput(onnx::ModelProto const&) in ir_pb_converter.cc.o
      ...
```

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 20.04*): macOS 13, 14, 15 and 26
- ONNX version (*e.g. 1.13*): 1.18.0, 1.19.0 and 075e319
- Python version: 3.9 and 3.13
- GCC/Compiler version (if compiling from source): Apple Clang 17.0.0 (macOS 26) and others for each OS
- CMake version: 4.1.2
- Protobuf version: 32.1, 29.5 and `ONNX_BUILD_CUSTOM_PROTOBUF=ON`


### Reproduction instructions
Originally seen at Homebrew when updating formula, but was able to reproduce off a git clone on macOS with just:

`cmake -S . -B build -DBUILD_SHARED_LIBS=ON && cmake --build build`

### Expected behavior
Successful build

### Notes
Seems to be another symbol visibility issue.

As an experiment, in 1.19.0, build succeed if I remove:
https://github.com/onnx/onnx/blob/57b9c6a4f6eebb09ae1b34cfb632078518f72832/CMakeLists.txt#L424

and in main if I remove:
https://github.com/onnx/onnx/blob/075e31993ae5d29659a10e8f3678c0e8731ae0b4/CMakeLists.txt#L412

Cross referencing Homebrew PRs where error was first seen:
- https://github.com/Homebrew/homebrew-core/pull/223395
- https://github.com/Homebrew/homebrew-core/pull/235202

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -418,7 +418,6 @@ add_onnx_global_defines(onnx_proto)
 add_library(onnx ${ONNX_SRCS} ${ONNX_PROTO_SRCS})
 add_dependencies(onnx onnx_proto)
 set_target_properties(onnx PROPERTIES CXX_VISIBILITY_PRESET hidden)
-set_target_properties(onnx PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
 add_onnx_global_defines(onnx)
 
 if(ONNX_BUILD_PYTHON)

--- a/onnx/checker.h
+++ b/onnx/checker.h
@@ -158,24 +158,22 @@ void check_function(const FunctionProto& function, const CheckerContext&, const 
 // Check schema compatibility for 2 opset versions for a given node.
 // Checks whether the schema for 2 versions is same, this is true when the opschema
 // does not change between versions.
-void check_opset_compatibility(
+ONNX_API void check_opset_compatibility(
     const NodeProto& node,
     const CheckerContext& ctx,
     const std::unordered_map<std::string, int>& func_opset_imports,
     const std::unordered_map<std::string, int>& model_opset_imports);
 
 // Checks all model local functions present in ModelProto
-void check_model_local_functions(
-    const ModelProto& model,
-    const CheckerContext& ctx,
-    const LexicalScopeContext& parent_lex);
+ONNX_API void
+check_model_local_functions(const ModelProto& model, const CheckerContext& ctx, const LexicalScopeContext& parent_lex);
 
-void check_model(
+ONNX_API void check_model(
     const ModelProto& model,
     bool full_check = false,
     bool skip_opset_compatibility_check = false,
     bool check_custom_domain = false);
-void check_model(
+ONNX_API void check_model(
     const std::string& model_path,
     bool full_check = false,
     bool skip_opset_compatibility_check = false,
@@ -184,7 +182,7 @@ std::string resolve_external_data_location(
     const std::string& base_dir,
     const std::string& location,
     const std::string& tensor_name);
-bool check_is_experimental_op(const NodeProto& node);
+ONNX_API bool check_is_experimental_op(const NodeProto& node);
 
 } // namespace checker
 } // namespace ONNX_NAMESPACE

--- a/onnx/common/ir_pb_converter.h
+++ b/onnx/common/ir_pb_converter.h
@@ -43,7 +43,7 @@ void ExportModelProto(ModelProto* p_m, const std::shared_ptr<Graph>& g);
 
 std::unique_ptr<Graph> ImportModelProto(const ModelProto& mp);
 
-ModelProto PrepareOutput(const ModelProto& mp_in);
+ONNX_API ModelProto PrepareOutput(const ModelProto& mp_in);
 
 void assertNonNull(const std::shared_ptr<Graph>& g);
 } // namespace ONNX_NAMESPACE

